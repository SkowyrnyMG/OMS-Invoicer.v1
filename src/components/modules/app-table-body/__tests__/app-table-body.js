import React from 'react';

import {
  renderWithReduxRouter,
  screen,
  fireEvent,
  leftClick,
  cleanup,
} from 'utils/tests/test-helper';
import AppTableBody from '../app-table-body';

const TEST_COLUMNS = [
  {
    Header: 'TEST1',
    accessor: 'test1',
  },
  {
    Header: 'TEST2',
    accessor: 'test2',
  },
  {
    Header: 'TEST3',
    accessor: 'test3',
  },
];

const TEST_DATA = [
  {
    test1: 'test1 value',
    test2: 'test2 value',
    test3: 'test3 value',
  },
];

const TEST_DEFAULT_COLUMN = {
  minWidth: 20,
  width: 20,
  maxWidth: 20,
};

const sortArrowCheck = (
  tableHeader,
  upArrowNode,
  downArrowNode,
  upArrowSvg,
  downArrowSvg,
) => {
  let upArrowNodeClone = upArrowNode;
  let downArrowNodeClone = downArrowNode;

  expect(upArrowNodeClone).toBeNull();
  expect(downArrowNodeClone).toBeNull();

  fireEvent.click(tableHeader, leftClick);
  upArrowNodeClone = screen.queryByText(upArrowSvg);
  downArrowNodeClone = screen.queryByText(downArrowSvg);
  expect(upArrowNodeClone).toBeInTheDocument();
  expect(downArrowNodeClone).toBeNull();

  fireEvent.click(tableHeader, leftClick);
  upArrowNodeClone = screen.queryByText(upArrowSvg);
  downArrowNodeClone = screen.queryByText(downArrowSvg);
  expect(downArrowNodeClone).toBeInTheDocument();
  expect(upArrowNodeClone).toBeNull();

  fireEvent.click(tableHeader, leftClick);
  upArrowNodeClone = screen.queryByText(upArrowSvg);
  downArrowNodeClone = screen.queryByText(downArrowSvg);
  expect(downArrowNodeClone).toBeNull();
  expect(upArrowNodeClone).toBeNull();
};

describe('AppTableBody', () => {
  test('should display in the document with correct styles and correct data in columns', () => {
    renderWithReduxRouter(
      <AppTableBody
        columns={TEST_COLUMNS}
        data={TEST_DATA}
        defaultColumn={TEST_DEFAULT_COLUMN}
        setCurrentPosValues={(row) => console.log(row)}
      />,
    );

    const AppTableBodyComponent = screen.getByTestId('app-table-body');
    const TableBody = screen.getByRole('row', {
      name: 'test1 value test2 value test3 value',
    });

    expect(AppTableBodyComponent).toBeInTheDocument();
    expect(AppTableBodyComponent).toMatchSnapshot();
    expect(TableBody).toBeInTheDocument();
  });

  test('should not display any table rows when data is not provided', () => {
    renderWithReduxRouter(
      <AppTableBody
        columns={TEST_COLUMNS}
        data={[]}
        defaultColumn={TEST_DEFAULT_COLUMN}
        setCurrentPosValues={(row) => console.log(row)}
      />,
    );

    const TableBody = screen.queryByRole('row', {
      name: 'test1 value test2 value test3 value',
    });

    expect(TableBody).toBeNull();
  });

  test('should conditionaly display sort arrows next to th name when clicked', () => {
    renderWithReduxRouter(
      <AppTableBody
        columns={TEST_COLUMNS}
        data={TEST_DATA}
        defaultColumn={TEST_DEFAULT_COLUMN}
        setCurrentPosValues={(row) => console.log(row)}
      />,
    );
    const firstTableHeader = screen.getByText('TEST1');
    const secondTableHeader = screen.getByText('TEST2');
    const thirdTableHeader = screen.getByText('TEST3');
    const upArrow = screen.queryByText('up-arrow-icon.svg');
    const downArrow = screen.queryByText('down-arrow-icon.svg');

    sortArrowCheck(
      firstTableHeader,
      upArrow,
      downArrow,
      'up-arrow-icon.svg',
      'down-arrow-icon.svg',
    );

    sortArrowCheck(
      secondTableHeader,
      upArrow,
      downArrow,
      'up-arrow-icon.svg',
      'down-arrow-icon.svg',
    );

    sortArrowCheck(
      thirdTableHeader,
      upArrow,
      downArrow,
      'up-arrow-icon.svg',
      'down-arrow-icon.svg',
    );
    cleanup();
  });
});
