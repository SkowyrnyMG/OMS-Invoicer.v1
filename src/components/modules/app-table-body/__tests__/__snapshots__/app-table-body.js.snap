// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTableBody should display in the document with correct styles and correct data in columns 1`] = `
.c4 {
  height: 60vh;
  width: 100%;
  background-color: #FFFFFF;
  border-radius: 0.5rem;
  box-shadow: 0 0 10px -4px rgba(0,0,0,0.25);
  overflow: scroll;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 2.5rem;
  height: 5rem;
  background: #FFFFFF;
  border-radius: 0.5rem;
  box-shadow: 0 0 10px -4px rgba(0,0,0,0.25);
}

.c2 {
  margin-left: 2.5rem;
  height: 2rem;
  fill: #8F8F8F;
}

.c3 {
  margin-left: 1rem;
  width: 50%;
  font-size: 20px;
  outline: none;
  border: none;
  border-bottom: 2px solid #F0F0FD;
}

.c5 {
  border-spacing: 0;
  border-collapse: separate;
}

.c5 thead {
  background: #446DF6;
  color: #FFFFFF;
  border: none;
}

.c5 td:not(:last-child),
.c5 thead tr th:not(:last-child) {
  padding: 0.2rem;
  border-right: 1px solid #F0F0FD;
}

.c5 td,
.c5 th {
  position: relative;
  cursor: pointer;
}

.c5 th {
  position: relative;
}

.c5 th span {
  position: absolute;
  right: 2%;
  top: 50%;
  -webkit-transform: translateY(-35%);
  -ms-transform: translateY(-35%);
  transform: translateY(-35%);
}

.c5 td {
  padding: 0.3rem 1rem !important;
  overflow-x: hidden;
}

.c5 tbody * {
  border-bottom: 1px solid #F0F0FD !important;
}

.c5 tr .cancel {
  color: red !important;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 2rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c8 {
  background: none;
  border: none;
  outline: none;
  cursor: pointer;
}

.c8:disabled {
  pointer-events: none;
}

.c8 svg {
  padding: 0 0.5rem;
  width: 2.5rem;
  height: 1.5rem;
  fill: #446DF644;
  -webkit-transition: 0.25s all;
  transition: 0.25s all;
}

.c8:hover svg {
  fill: #446DF6;
}

.c8:active svg {
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 1rem;
  padding: 0.25rem;
  border: 1px solid #446DF644;
  border-radius: 0.5rem;
  -webkit-transition: 0.25s all;
  transition: 0.25s all;
}

.c9::after {
  content: '';
  position: absolute;
  display: block;
  right: 5%;
  top: 50%;
  width: 1rem;
  height: 1rem;
  background: #446DF644;
  -webkit-clip-path: polygon(0 0,100% 0,50% 100%);
  clip-path: polygon(0 0,100% 0,50% 100%);
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  z-index: 300;
  -webkit-transition: 0.25s all;
  transition: 0.25s all;
}

.c9:hover {
  border-color: #446DF6;
}

.c9:hover::after {
  background: #446DF6;
}

.c9:active ::after {
  background: #000;
}

.c10 {
  padding: 0.1rem;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  outline: none;
}

.c10 option * {
  width: 100%;
}

.c6 {
  position: absolute;
  display: block;
  width: 30px;
  height: 100%;
  top: 0;
  right: 0;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
  z-index: 5000 !important;
}

@media (max-width:767px) {
  .c4 {
    height: 60vh;
  }
}

<div
  class="c0"
  data-testid="app-table-body"
>
  <div
    class="c1"
  >
    <svg
      class="c2"
    >
      search-icon.svg
    </svg>
    <input
      class="c3"
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  <div
    class="c4"
  >
    <table
      class="c5"
      role="table"
    >
      <thead>
        <tr
          role="row"
          style="display: flex; width: 60px;"
        >
          <th
            colspan="1"
            role="columnheader"
            style="display: inline-block; box-sizing: border-box; width: 20px; position: relative; cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="c6"
              draggable="false"
              role="separator"
              style="cursor: col-resize;"
            />
            TEST1
            <span />
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="display: inline-block; box-sizing: border-box; width: 20px; position: relative; cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="c6"
              draggable="false"
              role="separator"
              style="cursor: col-resize;"
            />
            TEST2
            <span />
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="display: inline-block; box-sizing: border-box; width: 20px; position: relative; cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="c6"
              draggable="false"
              role="separator"
              style="cursor: col-resize;"
            />
            TEST3
            <span />
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
          style="display: flex; width: 60px; opacity: 1;"
        >
          <td
            role="cell"
            style="display: inline-block; box-sizing: border-box; width: 20px;"
          >
            test1 value
          </td>
          <td
            role="cell"
            style="display: inline-block; box-sizing: border-box; width: 20px;"
          >
            test2 value
          </td>
          <td
            role="cell"
            style="display: inline-block; box-sizing: border-box; width: 20px;"
          >
            test3 value
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="c7"
  >
    <div>
      <button
        class="c8"
        disabled=""
      >
        <svg>
          larr-icon.svg
        </svg>
      </button>
      <span>
        <strong>
           1 / 25  of 25
        </strong>
      </span>
      <button
        class="c8"
        disabled=""
      >
        <svg>
          rarr-icon.svg
        </svg>
      </button>
    </div>
    <div
      class="c9"
    >
      <select
        class="c10"
      >
        <option
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
      </select>
    </div>
  </div>
</div>
`;
